### ISSUE TITLE
`quotes`: allowTemplateLiterals false does not work properly

### ISSUE THREAD

**Issue by Zamiell** [eslint-stylistic/eslint-stylistic#542]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

#### eslint.config.mjs

```js
import ESLintPluginStylistic from "@stylistic/eslint-plugin";

export default {
  plugins: {
    "@stylistic": ESLintPluginStylistic,
  },

  rules: {
    "@stylistic/quotes": [
      "warn",
      "double",
      {
        avoidEscape: true,
        allowTemplateLiterals: false,
      },
    ],
  },
};
```

#### main.js

```
const a = `"a"`;
```

- The rule does not warn on this line, which is a bug.
- It should auto-fix it to: `const a = '"a"'`
- I think this is a regression, because this case seemed to be correctly handled when I used `@typescript-eslint/quotes` (before they deprecated it).

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests
